third_party/amd/lib/TritonAMDGPUTransforms/AccelerateAMDMatmul.cpp

```diff
@@ -190,6 +190,50 @@ class BlockedToMFMA : public mlir::RewritePattern {
     return {nonKDim, kDim};
   }
 
+  Value convertAndPromoteDotOperand(mlir::PatternRewriter &rewriter,
+                                    Value oldDotOperand,
+                                    ::mlir::Attribute encoding,
+                                    Type promotedElemType) const {
+    assert(promotedElemType.isIntOrFloat());
+
+    auto loc = oldDotOperand.getLoc();
+    auto oldType = oldDotOperand.getType().cast<RankedTensorType>();
+    auto oldElemType = oldType.getElementType();
+
+    assert(oldElemType.isIntOrFloat());
+    assert(oldElemType.isIntOrIndex() == promotedElemType.isIntOrIndex());
+
+    auto convertedType =
+        RankedTensorType::get(oldType.getShape(), oldElemType, encoding);
+
+    Value convertedDotOperand = rewriter.create<ttg::ConvertLayoutOp>(
+        loc, convertedType, oldDotOperand);
+
+    if (promotedElemType == oldElemType)
+      return convertedDotOperand;
+
+    Type promotedType = convertedType.cloneWith(std::nullopt, promotedElemType);
+
+    Value promotedDotOperand;
+
+    if (promotedElemType.isIntOrIndex()) {
+      // TODO Implement integer casting
+      assert(!promotedElemType.isIntOrIndex() &&
+             "need to implement integer promotion");
```

Comment:
Are `if` and `assert` here doing the same thing?
If you want to keep the if .. else structure, maybe use unreachable or report_fatal to remind us this is not implemented yet.

Reply:
In cases something is not supported yet, shouldn't we just skip mfma transformation?

Reply:
When I wrote this code I was not sure if we need to cast integer values (we do not have any related tests at the moment)

I need to clarify this question, but at the same time I want to fix tests ASAP. So I left a stub.